features/Context/FieldTypeFormContext.php

```diff
@@ -144,23 +145,120 @@ public function theFieldDefinitionIsMarkedAsRequired()
+        $this->theFieldShouldBeFlaggedAsRequired('input');
+    }
+
+    /**
+     * @Given /^the field definition is set to single choice$/
+     */
+    public function theFieldDefinitionIsSetToSingleChoice()
+    {
+        $this->setBooleanFieldDefinitionOption('isMultiple', false);
+    }
+
+    /**
+     * @Given /^the field definition is set to multiple choice$/
+     */
+    public function theFieldDefinitionIsSetToMultipleChoice()
+    {
+        $this->setBooleanFieldDefinitionOption('isMultiple', true);
+    }
+
+    /**
+     * @Then it should contain a select field
+     */
+    public function itShouldContainASelectField()
+    {
+        $this->assertElementOnPage(
+            sprintf(
+                'div.ezfield-identifier-%s fieldset select',
+                self::$fieldIdentifier
+            )
+        );
+    }
+
+    /**
+     * @Then the select field should be flagged as required
+     */
+    public function theSelectFieldShouldBeFlaggedAsRequired()
+    {
+        $this->theFieldShouldBeFlaggedAsRequired('select');
+    }
+
+    /**
+     * Assert that a form field should be flagged as required.
+     *
+     * @param $elementType string The type of form element (e.g. 'input', 'select')
+     */
+    protected function theFieldShouldBeFlaggedAsRequired($elementType)
+    {
+        $nodeElements = $this->getSession()->getPage()->findAll(
             'css',
-            sprintf('div.ezfield-identifier-%s fieldset input', self::$fieldIdentifier)
+            sprintf('div.ezfield-identifier-%s fieldset %s', self::$fieldIdentifier, $elementType)
         );
 
-        Assertion::assertNotEmpty($inputNodeElements, 'The input field is not marked as required');
+        Assertion::assertNotEmpty($nodeElements, "The $elementType field is not marked as required");
+
+        foreach ($nodeElements as $nodeElement) {
+            $failureMessage = trim(sprintf(
+                '%s %s with id %s is not flagged as required',
+                $nodeElement->hasAttribute('type') ? $nodeElement->getAttribute('type') : '',
+                $elementType,
+                $nodeElement->getAttribute('id')
+            ));
 
-        foreach ($inputNodeElements as $inputNodeElement) {
             Assertion::assertEquals(
                 'required',
-                $inputNodeElement->getAttribute('required'),
-                sprintf(
-                    '%s input with id %s is not flagged as required',
-                    $inputNodeElement->getAttribute('type'),
-                    $inputNodeElement->getAttribute('id')
-                )
+                $nodeElement->getAttribute('required'),
+                $failureMessage
             );
         }
     }
+
+    /**
+     * Set a boolean field definition option $option to $value.
+     *
+     * @param $option string The field definition option
+     * @param $value bool The option value
+     */
+    protected function setBooleanFieldDefinitionOption($option, $value)
```

Comment:
Is it limited to boolean fields ? It would work with any field with the current code, would it not ?

I'd also make this method public, since it can be used by other contexts. I have drafted an example we might want to use:

```
<?php
namespace EzSystems\RepositoryForms\Features\Context;

use Behat\Behat\Context\SnippetAcceptingContext;
use Behat\Behat\Hook\Scope\BeforeScenarioScope;
use Behat\MinkExtension\Context\RawMinkContext;

class SelectionFieldTypeFormContext extends RawMinkContext implements SnippetAcceptingContext
{
    /**
     * @var \EzSystems\RepositoryForms\Features\Context\FieldTypeFormContext
     */
    private $fieldTypeFormContext;

    /** @BeforeScenario */
    public function gatherContexts(BeforeScenarioScope $scope)
    {
        $this->fieldTypeFormContext = $scope->getEnvironment()->getContext(
            'EzSystems\RepositoryForms\Features\Context\FieldTypeFormContext'
        );
    }

    /**
     * @Given /^the field definition is set to single choice$/
     */
    public function setFieldDefinitionToSingleChoice()
    {
        $this->fieldTypeFormContext->setFieldDefinitionOption('isMultiple', false);
    }

    /**
     * @Given /^the field definition is set to multiple choice$/
     */
    public function setFieldDefinitionToMultipleChoice()
    {
        $this->fieldTypeFormContext->setFieldDefinitionOption('isMultiple', true);
    }
}
```

(It does imply a renaming of the method to `setFieldDefinitionOption`, but the idea is the same)